src/parser/type_checking.c

```diff
@@ -19665,6 +19665,12 @@ pt_fold_const_expr (PARSER_CONTEXT * parser, PT_NODE * expr, void *arg)
   const char *alias_print;
   unsigned is_hidden_column;
 
+  // in case of error this functions return the unmodified expr node
+  // to avoid a memory leak we need to restore the link of expr->next to expr_next
+  // this will be done in the goto end label
+  bool has_error_so_far = false;
```

Comment:
Please rename has_error_so_far just as has_error.

Also replace print_pt_error with was_error_set (and initialize as false and set to true when it is set).